geniza/corpus/management/commands/import_data.py

```diff
@@ -336,3 +372,153 @@ def log_creation(self, *objects):
+        for date in all_dates:
+            try:
+                dates.append(parse(date, default=DEFAULT_EVENT_DATE).date())
+            except ParserError:
+                continue
+
+        # make sure we have same number of users/dates by padding with None
+        while len(users) < len(dates):
+            users.insert(0, None)
+
+        # moving backwards in time, pair dates with users and event types
+        events = []
+        for i in range(len(dates) - 1, -1, -1):
+            date = dates[i]
+
+            # final (earliest) date is creation; all others are revisions
+            event_type = "Created" if i == 0 else "Revised"
+
+            # if we have a date without a user, assign to the whole team
+            user = users[i] or self.team_user
+
+            # if it was two users together, log two separate events because
+            # each event must have exactly one user
+            if type(user) == tuple:
+                events.append(
+                    {"type": event_type, "user": user[0], "date": date})
+                events.append(
+                    {"type": event_type, "user": user[1], "date": date})
+                logging.debug(
+                    f"Found coauthored event for PGPID {pgpid}: {events[-2:]}")
+                continue
+
+            # otherwise just create a single event for this user
+            events.append({"type": event_type, "user": user, "date": date})
+            logging.debug(f"Found event for PGPID {pgpid}: {events[-1]}")
+
+        # sort chronologically and return
+        events.sort(key=itemgetter("date"))
+        return events
+
+    def log_edit_history(self, doc, events):
+        """Given a Document and a sequence of events from get_edit_history,
+        create corresponding Django log entries to represent that history.
+
+        Always creates an entry by the script user (`SCRIPT_USERNAME`) with a
+        timestamp of now to mark the import event itself.
+
+        Optional `events` is list of dict from get_edit_history. Each event has
+        a type (creation or revision), associated user, and timestamp.
+        """
+
+        # for each historic event, create a corresponding django log entry.
+        # we use objects.create() instead of the log_action helper so that we
+        # can control the timestamp.
+        for event in events:
+            dt = datetime(year=event["date"].year,
+                          month=event["date"].month,
+                          day=event["date"].day)
+            LogEntry.objects.create(
+                user=event["user"],                            # FK in django
+                content_type=self.content_types[Document],     # FK in django
+                object_id=str(doc.pk),         # TextField in django
+                object_repr=str(doc)[:200],    # CharField with limit in django
+                change_message=event["type"],
+                action_flag=ADDITION if event["type"] == "Created" else CHANGE,
```

Comment:
could we just use the django flags in the event type and have th legacy input/change message keyed on that flag?

Reply:
in my mind, this function deals with how to represent the events created in the other function in django, so the logic of which flag matches which event belongs here rather than in the other function. that's also why the other function uses `date` instead of `datetime`.